### PR TITLE
feat(kstyles): multi-line truncation [khcp-3552]

### PR DESCRIPTION
**Pull request by kaiarrowood**
# Summary

- Add support for `.multi-line-truncation`
- Deprecate `.truncate-multi`

image

## Vue 3

**Did you create a corresponding PR to add this change to the `beta` branch? (required)**

- [ ] **Yes**, here is a link to the PR: `[link to beta branch PR]`
- [ ] **No**, I did not create a PR for the `beta` branch and I have added the `port to beta branch` tag to this PR.
- [ ] **No**, because `[type your reasons]`

If you have questions, tag `@adamdehaven` or `@kaiarrowood`.

---

## PR Checklist

- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: docs/style-guide/utilities.md [hunk 1]
```diff
@@ -171,6 +171,33 @@ For each size in our [type definitions](/style-guide/type.html) we include a uti
 
 We support both single line truncation as well as multi-line. Multi-line truncation is customizable with the following properties:
 
+- `MaxLineLimit` - the maximum number of allowed lines (defaults to `3`)
```
Comment (adamdehaven) [inline docs/style-guide/utilities.md:174]: question: Should this CSS variable be named something to indicate it's related to truncation? 

Also, in the docs page, I don't see a mention that this is a CSS variable except in the example?

File: packages/styles/_utils.scss [hunk 2]
```diff
@@ -165,6 +165,12 @@ $sides: (top, right, bottom, left);
     right: var(--TPosRight, 12px);
   }
 }
+.multi-line-truncation {
+  display: -webkit-box;
```
Comment (adamdehaven) [inline packages/styles/_utils.scss:169]: Doesn't this completely break if you're not in a WebKit browser?

Comment (kaiarrowood) [reply packages/styles/_utils.scss:169]: image

Comment (kaiarrowood) [reply packages/styles/_utils.scss:169]: We don't care about IE

**Review by adamdehaven — APPROVED** [review]
LGTM. Did more research and this should be good

**Comment by adamdehaven** [comment]
image

Props to @mihai-peteu for finding this gem

**Review by adamdehaven — APPROVED** [review]
Approving so we can implement where needed; need to verify this works in all modern browsers, not just webkit

**Comment by kaiarrowood** [comment]
Verification:

Edge:

image

Firefox:

image


@adamdehaven I'll need you to verify it functions in Safari

**Pull request merged**